src/core/server/saved_objects/saved_objects_service.ts

```diff
@@ -435,6 +436,7 @@ export class SavedObjectsService
       // and the promise above fulfils as undefined. We shouldn't trigger migrations at that point.
       if (compatibleNodes) {
         this.logger.info('Starting saved objects migrations');
+        await deleteIndexTemplates({ client: client.asInternalUser, log: this.logger });
         await migrator.runMigrations();
```

Comment:
I had to put this here to properly await `compatibleNodes`, meaning that it will be skipped as the migration is if `skipMigrations` is `true`. But given that this should be considered part of the migration, I think this is fine.